src/token.cc

```diff
@@ -233,9 +233,11 @@ void expr_t::token_t::next(std::istream& in, const parse_flags_t& pflags)
     in.get(c);
     amount_t temp;
     temp.parse(in, PARSE_NO_MIGRATE);
-    in.get(c);
-    if (c != '}')
-      expected('}', c);
+    int i = in.get();
+    if (i != '}')
+      expected('}', i);
+    else
+      c = i & 0xff;
```

Comment:
What does this change do?  If I understand correctly, it just leads to c becoming 255/-1 depending on whether `char` is `unsigned`/`signed`?  But c is not used afterwards anyway?

Reply:
The `expected()` method apparently relies on getting a -1 to indicate EOF. But `in.get(c)` does not promise to set `c` (at all) in the case of EOF, as far as I could determine, whereas `i = in.get()` (where `i` is `int` rather than `char`) of course does. So, that was the intent of the change.

As for whether `c` is used again afterwards, you're right: we could get rid of that last part. In the original code, `c` does end up holding the value of the character in the non-EOF case. So this change as written preserves that behavior, which makes it easier for a reviewer to read without looking beyond this localized context. I wouldn't mind removing those last couple lines.

Reply:
I'd drop any code which does not serve a useful purpose.  In the end, it will just clutter the file and make it even less obvious to future readers what a function is supposed to be doing.